programs/cleos/main.cpp

```diff
@@ -2247,6 +2249,76 @@ struct closerex_subcommand {
    }
 };
 
+struct activate_subcommand {
+   string feature_name_str;
+
+   activate_subcommand(CLI::App* actionRoot) {
+      auto activate = actionRoot->add_subcommand("activate", localized("Activate protocol feature by name"));
+      activate->add_option("feature",  feature_name_str,  localized("The name, one of below (lowercase also works):\n"
+                       "ONLY_LINK_TO_EXISTING_PERMISSION\n"
```

Comment:
Can we have a way of generating these from parsing the protocol_feature_manager.cpp? We have this pattern in quite a few places as is where we have hard-coded the protocol feature names and digests.  It ultimately makes it incredibly difficult to add new protocol features as all the places you have to go and fix are not connected and usually won't break certain tests, etc.

Reply:
Thanks @larryk85 and @spoonincode I will call `get_supported_protocol_features` to get a list. This code will never change again when protocol features change.